httpbakery/error.go

```diff
@@ -275,24 +269,30 @@ func newDischargeRequiredErrorWithVersion(m *macaroon.Macaroon, path string, ori
 // header set to "Macaroon".
 const BakeryProtocolHeader = "Bakery-Protocol-Version"
 
-// versionFromRequest determines the bakery protocol version from a client
-// request. If the protocol cannot be determined, or is invalid,
-// the original version of the protocol is used.
-func versionFromRequest(req *http.Request) version {
+// RequestVersion determines the bakery protocol version from a client
+// request. If the protocol cannot be determined, or is invalid, the
+// original version of the protocol is used. If a later version is
+// found, the latest known version is used, which is OK because versions
+// are backwardly compatible.
+//
+// TODO as there are no known version 0 clients, default to version 1
```

Comment:
Because that would affect a bunch of the tests and this PR is way too big already...